src/components/SearchBar/index.jsx

```diff
@@ -22,42 +23,66 @@ const DesktopSearchBarContainer = styled.div`
   `};
 `;
 
-function SearchBar() {
+function SearchBar({ onSearch }) {
   const [location, setLocation] = useState('');
-  const [status, useStatus] = useState(false);
+  const [status, setStatus] = useState(false);
   const [title, setTitle] = useState('');
 
-  // @TODO Implement search and filter functionality.
-  const handleOnChangeByLocation = () => {};
-  const handleOnChangeByStatus = () => {};
-  const handleOnChangeByTitle = () => {};
-  const handleOnSearch = () => {};
+  const handleOnChangeByLocation = (e) => setLocation(e.target.value);
+
+  const handleOnChangeByStatus = () => setStatus(!status);
+
+  const handleOnChangeByTitle = (e) => setTitle(e.target.value);
+
+  const handleOnSearch = () => {
+    const filtered = jobsData.filter((job) => {
```

Comment:
You are using `jobsData` instead of the `jobs` state value. Can you please update this so the search is filtered by the jobs that are currently available for view to the users? Thank you!

Reply:
Updated the component to use `jobs` instead of `jobsData`